internal/anchor/reconciler.go

```diff
@@ -113,8 +114,15 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	r.updateState(log, inst, snsInst)
 
 	// Handle the case where the anchor is being deleted.
-	if deleting, err := r.onDeleting(ctx, log, inst, snsInst); deleting {
-		return ctrl.Result{}, err
+	if inst.DeletionTimestamp.IsZero() {
+		// Add finalizers on all non-forbidden anchors to ensure it's not deleted until
+		// after the subnamespace is deleted.
+		if !controllerutil.ContainsFinalizer(inst, api.MetaGroup) {
+			controllerutil.AddFinalizer(inst, api.MetaGroup)
+		}
+	} else {
+		// Stop reconciliation as anchor is being deleted
+		return ctrl.Result{}, r.onDeleting(ctx, log, inst, snsInst)
 	}
```

Comment:
Can you please re-order the statements to have the early-exit come first? Also, `AddFinalizer` is idempotent so you don't need to do the check first.

```
	// Handle the case where the anchor is being deleted.
	if !inst.DeletionTimestamp.IsZero() {
		return ctrl.Result{}, r.onDeleting(ctx, log, inst, snsInst)
	}

	// Add finalizers on all non-forbidden anchors to ensure it's not deleted until
	// after the subnamespace is deleted.
	controllerutil.AddFinalizer(inst, api.MetaGroup)
```

Reply:
I will invert the if. Going to add a conditional update in my next PR, but I'll remove the conditional around `AddFinalizer` for now. :-)

Reply:
Fixed now, PTAL!